rskj-core/src/test/java/co/rsk/peg/BridgeSupportAddSignatureTest.java

```diff
@@ -0,0 +1,727 @@
+        }
+    }
+
+    /**
+     * Helper method to test addSignature() with a valid federatorPublicKey parameter and both valid/invalid signatures
+     *
+     * @param privateKeysToSignWith keys used to sign the tx. Federator key when we want to produce a valid signature, a random key when we want to produce an invalid signature
+     * @param numberOfInputsToSign  There is just 1 input. 1 when testing the happy case, other values to test attacks/bugs.
+     * @param signatureCanonical    Signature should be canonical. true when testing the happy case, false to test attacks/bugs.
+     * @param signTwice             Sign again with the same key
+     * @param expectedResult        "InvalidParameters", "PartiallySigned" or "FullySigned"
+     */
+    private void addSignatureFromValidFederator(List<BtcECKey> privateKeysToSignWith, int numberOfInputsToSign, boolean signatureCanonical, boolean signTwice, String expectedResult) throws Exception {
+        // Federation is the genesis federation ATM
+        Federation federation = bridgeConstantsRegtest.getGenesisFederation();
+        Repository repository = createRepository();
+
+        final Keccak256 keccak256 = PegTestUtils.createHash3();
+
+        Repository track = repository.startTracking();
+        BridgeStorageProvider provider = new BridgeStorageProvider(track, PrecompiledContracts.BRIDGE_ADDR, bridgeConstantsRegtest, activationsBeforeForks);
+
+        BtcTransaction prevTx = new BtcTransaction(btcRegTestParams);
+        TransactionOutput prevOut = new TransactionOutput(btcRegTestParams, prevTx, Coin.FIFTY_COINS, federation.getAddress());
+        prevTx.addOutput(prevOut);
+
+        BtcTransaction t = new BtcTransaction(btcRegTestParams);
+        TransactionOutput output = new TransactionOutput(btcRegTestParams, t, Coin.COIN, new BtcECKey().toAddress(btcRegTestParams));
+        t.addOutput(output);
+        t.addInput(prevOut).setScriptSig(createBaseInputScriptThatSpendsFromTheFederation(federation));
+        provider.getRskTxsWaitingForSignatures().put(keccak256, t);
+        provider.save();
+        track.commit();
+
+        track = repository.startTracking();
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        List<LogInfo> logs = new ArrayList<>();
+        BridgeEventLogger eventLogger = new BrigeEventLoggerLegacyImpl(bridgeConstantsRegtest, activations, logs);
+
+        BridgeSupport bridgeSupport = bridgeSupportBuilder
+                .withBridgeConstants(bridgeConstantsRegtest)
+                .withProvider(new BridgeStorageProvider(
+                        track,
+                        contractAddress,
+                        bridgeConstantsRegtest,
+                        activationsAfterForks
+                ))
+                .withRepository(track)
+                .withEventLogger(eventLogger)
+                .withExecutionBlock(mock(Block.class))
```

Comment:
BridgeSupportBuilder sets a mock by default for every param, so this isn't really needed